phat/contracts/brick_profile_factory/lib.rs

```diff
@@ -75,27 +76,88 @@ mod brick_profile_factory {
             Ok(())
         }
 
+        /// Import user profile addresses (only owner). This will overwrite existing addresses.
+        #[ink(message)]
+        pub fn import_user_profiles(
+            &mut self,
+            profiles: Vec<(AccountId, AccountId)>,
+        ) -> Result<()> {
+            self.ensure_owner()?;
+            self.users.extend(
+                profiles
+                    .into_iter()
+                    .map(|kv| (kv.0, BrickProfileRef::from_account_id(kv.1.into()))),
+            );
+            Ok(())
+        }
+
+        /// Instantiate a user profile contract for caller, overwrite the existing one.
+        /// Return old user profile contract if there is.
+        #[ink(message)]
+        pub fn force_create_user_profile(&mut self) -> Result<Option<AccountId>> {
+            let caller = self.env().caller();
+
+            let user_profile = self.instantiate_profile()?;
+
+            let old_user_profile = self
+                .users
+                .insert(caller, user_profile)
+                .map(|p| p.to_account_id());
+            self.nonce += 1;
+
+            Ok(old_user_profile)
+        }
+
         /// Instantiate a user profile contract for caller. Only once for each account.
         #[ink(message)]
         pub fn create_user_profile(&mut self) -> Result<()> {
             let caller = self.env().caller();
-
             if self.users.contains_key(&caller) {
                 return Err(Error::NoDuplicatedUserProfile);
             }
 
-            let random = signing::derive_sr25519_key(&self.user_count.to_be_bytes());
-            let user_profile = BrickProfileRef::new(caller)
-                .endowment(0)
-                .salt_bytes(&random[..4])
-                .code_hash(self.profile_code_hash)
-                .try_instantiate()
-                .map_err(|e| Error::FailedToCreateProfile(format!("{:?}", e)))?
+            self.force_create_user_profile()?;
+            Ok(())
+        }
+
+        /// Instantiate and setup a user profile contract for caller, overwrite the existing one.
+        /// This configures the js runner and generates the first evm account.
+        /// Return old user profile contract if there is.
+        #[ink(message)]
+        pub fn force_setup_user_profile(
+            &mut self,
+            js_runner: AccountId,
+            rpc: String,
+        ) -> Result<Option<AccountId>> {
+            let caller = self.env().caller();
+
+            let mut user_profile = self.instantiate_profile()?;
```

Comment:
Readability: consider moving `instantiate_profile()` and `nonce += 1` out so an internal function. Or maybe just inc the nonce in `instantiate_profile()`